analyzer/restapi-plugin/src/main/java/eu/fasten/analyzer/restapiplugin/mvn/api/PackageApi.java

```diff
@@ -144,24 +144,44 @@
                                        @PathVariable("pkg_ver") String version,
                                        @RequestParam(value = "artifactRepository", required = false) String artifactRepo,
                                        @RequestParam(required = false) Long releaseDate) {
-        try {
-            if (KnowledgeBaseConnector.kbDao.assertPackageExistence(packageName, version)) {
+        String result;
+        String url;
+        if (!KnowledgeBaseConnector.kbDao.assertPackageExistence(packageName, version)) {
+            try {
+                LazyIngestionProvider.ingestArtifactIfNecessary(packageName, version, artifactRepo, releaseDate);
+            } catch (IllegalArgumentException ex) {
+                return new ResponseEntity<>(ex.getMessage(), HttpStatus.BAD_REQUEST);     
+            } catch (IOException ex) {
+                return new ResponseEntity<>(ex.getMessage(), HttpStatus.INTERNAL_SERVER_ERROR);
+            }
+            return new ResponseEntity<>("Package version not found, but should be processed soon. Try again later", HttpStatus.CREATED);
+        }
+        switch (KnowledgeBaseConnector.forge) {
+            case "mvn": {
                 var groupId = packageName.split(Constants.mvnCoordinateSeparator)[0];
                 var artifactId = packageName.split(Constants.mvnCoordinateSeparator)[1];
-                var url = String.format("%smvn/%s/%s/%s_%s_%s.json", KnowledgeBaseConnector.rcgBaseUrl,
+                url = String.format("%smvn/%s/%s/%s_%s_%s.json", KnowledgeBaseConnector.rcgBaseUrl,
                         artifactId.charAt(0), artifactId, artifactId, groupId, version).replace("\\/", "/");
-                var result = MavenUtilities.sendGetRequest(url);
-                if (result == null) {
-                    return new ResponseEntity<>("Could not find the requested data at " + url, HttpStatus.NOT_FOUND);
-                }
-                return new ResponseEntity<>(result, HttpStatus.OK);
-            } else {
-                LazyIngestionProvider.ingestArtifactIfNecessary(packageName, version, artifactRepo, releaseDate);
-                return new ResponseEntity<>("Package version not found, but should be processed soon. Try again later", HttpStatus.CREATED);
+                break;
             }
-        } catch (IOException e) {
-            return new ResponseEntity<>(e.getMessage(), HttpStatus.INTERNAL_SERVER_ERROR);
+            case "pypi": {
+                url = String.format("%s/%s/%s/%s/%s/cg.json", KnowledgeBaseConnector.rcgBaseUrl + KnowledgeBaseConnector.forge + "/" + KnowledgeBaseConnector.forge,
```

Comment:
There is a bug here: KnowledgeBaseConnector.forge is included twice!

Reply:
Well, I intentionally included in the lima url  twice the forge, because for some reason, both on monster and on lima the rcg path includes two pypi directories. For example for the shap:0.34.0 revision the rcg url is the following: https://lima.ewi.tudelft.nl/pypi/pypi/callgraphs/s/shap/0.34.0/cg.json . (https://lima.ewi.tudelft.nl/pypi/callgraphs/s/shap/0.34.0/cg.json returns a 404 error)

Reply:
Ugh, I see. Well it's easy to adjust this in the PyCG deployment to write to a /pypi/pypi path as well. Or the Delft guys @mir-am  could adjust the path on lima to make things look a bit nicer :-)

Reply:
I think it's not nice to have such strong dependencies on file system arrangements in the code, though.

Reply:
The lima url differs for each language: 
* https://lima.ewi.tudelft.nl/pypi/pypi/callgraphs/a/azureml-contrib-interpret/1.0.85/cg.json
* https://lima.ewi.tudelft.nl/debian/callgraphs/4/4digits/buster/1.1.4-1%2Bb1/amd64/file.json
* https://lima.ewi.tudelft.nl/mvn/z/zabbix-api/zabbix-api_io.github.hengyunabc_0.0.2.json
So I do not know if we can avoid this on the REST-API. Maybe on lima we could use a unified format for all languages